common/src/main/java/com/zegelin/cassandra/exporter/collector/StorageServiceMBeanMetricFamilyCollector.java

```diff
@@ -144,6 +147,6 @@ private StorageServiceMBeanMetricFamilyCollector(final StorageServiceMBean stora
             metricFamilyStreamBuilder.add(new GaugeMetricFamily("cassandra_storage_filesystem_unallocated_bytes", null, fileStoreUnallocatedSpaceMetrics.build()));
         }
 
-        return metricFamilyStreamBuilder.build();
+        return metricFamilyStreamBuilder.build().filter(mf -> exclusions.contains(mf.name));
```

Comment:
I've yet to pull this branch and test it, but here it looks like it's calling `Set.contains(…)` with a `String` and yet the `Set` contains elements of type `Harvester.Exclusion`.

`Set.contains(…)` is defined as:
> Returns true if this set contains the specified element. More formally, returns true if and only if this set contains an element e such that `(o==null ? e==null : o.equals(e))`.

In other words, it'll end up calling `Harvester.Exclusion.equals(…)` with a `String` which will always return `false`.

Reply:
I'm also hesitant to do runtime filtering. A significant amount of work was done to ensure that the runtime performance of the exporter was optimised. 

This is okay for now, but I'll rethink the best way to do filtering in the future.

Reply:
I just tried the following, and it fails:

```
final Set<Exclusion> exclusions = new HashSet<>();

exclusions.add(Exclusion.create("hello_world"));

assertTrue(exclusions.contains("hello_world"));
```

Reply:
Yes, you are correct. I misinterpreted equals method of CollectorExclusion. I will update cod to fix it.